sphinxext/opengraph/__init__.py

```diff
@@ -106,50 +147,37 @@ def get_tags(
         tags["og:description"] = description
 
     # image tag
-    # Get basic values from config
+    # Get basic values from config or field list
+    # image_url = fields.pop("og:image", config["ogp_image"])
+    # ogp_image_alt = fields.pop("og:image:alt", config["ogp_image_alt"])
+
     if "og:image" in fields:
-        image_url = fields["og:image"]
-        ogp_use_first_image = False
-        ogp_image_alt = fields.get("og:image:alt")
-        fields.pop("og:image", None)
+        image_url = image_abs_url(
+            fields.pop("og:image"), config["ogp_site_url"], docname, app
+        )
+        image_alt = fields.pop("og:image:alt", None)
+    elif fields.get("ogp_use_first_image", config["ogp_use_first_image"]) and (
+        first_image := doctree.next_node(nodes.image)
+    ):
+        # Use page url, since the image uri at this point is relative to the output page
+        image_url = image_abs_url(first_image["uri"], page_url)
+        image_alt = first_image.get("alt", fields.pop("og:image:alt", None))
     else:
-        image_url = config["ogp_image"]
-        ogp_use_first_image = config["ogp_use_first_image"]
-        ogp_image_alt = fields.get("og:image:alt", config["ogp_image_alt"])
-
-    fields.pop("og:image:alt", None)
-
-    if ogp_use_first_image:
-        first_image = doctree.next_node(nodes.image)
-        if (
-            first_image
-            and Path(first_image.get("uri", "")).suffix[1:].lower() in IMAGE_MIME_TYPES
-        ):
-            image_url = first_image["uri"]
-            ogp_image_alt = first_image.get("alt", None)
+        #
+        image_url = image_abs_url(
+            config["ogp_image"], config["ogp_site_url"], docname, app
+        )
+        image_alt = config["ogp_image_alt"]
```

Comment:
Open for comments on this implementation I have a few ideas but still would like to hear anything else, currently uses walrus operator (would like to avoid that if possible, since currently, we support down to 3.6)